src/Microsoft.IdentityModel.Logging/IdentityModelTelemetryUtil.cs

```diff
@@ -0,0 +1,139 @@
+
+        /// <summary>
+        /// Get the string that represents the client SKU.
+        /// </summary>
+        public static string ClientSku =>
+#if NET45
+            "ID_NET45";
+#elif NET461
+            "ID_NET461";
+#elif NET472
+            "ID_NET472";
+#elif NETSTANDARD2_0
+            "ID_NETSTANDARD2_0";
+#endif
+
+        /// <summary>
+        /// Get the string that represents the client version.
+        /// </summary>
+        public static string ClientVer => typeof(IdentityModelTelemetryUtil).GetTypeInfo().Assembly.GetName().Version.ToString();
+
+        /// <summary>
+        /// Adds a key and its value to the collection of telemetry data.
+        /// </summary>
+        /// <param name="key"> The name of the telemetry.</param>
+        /// <param name="value"> The value of the telemetry.</param>
+        /// <returns> true if the key is successfully added; otherwise, false.</returns>
+        public static bool AddTelemetryData(string key, string value)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                LogHelper.LogArgumentNullException(nameof(key));
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(value))
+            {
+                LogHelper.LogArgumentNullException(nameof(value));
+                return false;
+            }
+
+            if (defaultTelemetryValues.Contains(key))
+            {
+                LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.MIML10003, defaultTelemetryValues)));
+                return false;
+            }
+
+            telemetryData[key] = value;
+            return true;
+        }
+
+        /// <summary>
+        /// Removes a key and its value from the collection of telemetry data.
+        /// </summary>
+        /// <param name="key"> The name of the telemetry.</param>
+        public static void RemoveTelemetryData(string key)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw LogHelper.LogArgumentNullException(nameof(key));
+
+            if (defaultTelemetryValues.Contains(key))
+                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.MIML10003, defaultTelemetryValues)));
+
+            telemetryData.TryRemove(key, out _);
+        }
+
+        internal static void SetTelemetryData(HttpRequestMessage request)
+        {
+            if (request == null)
+                return;
+
+            foreach (var parameter in telemetryData)
+            {
+                // remove this header if it already exists.
+                // we don't want to add an additional value in case when a telemetry header already exists, but to overwrite it.
+                request.Headers.Remove(parameter.Key);
+                request.Headers.Add(parameter.Key, parameter.Value);
+            }
+        }
+
+        internal static void UpdateDefaultTelemetryData(string key, string value)
```

Comment:
Is this one still needed? Can you operate on telemetryData (also internal) in tests?

Reply:
Are you saying we can directly add to the dictionary from SAL, instead of using this method?

Reply:
Ah, thought that this was a leftover used only for tests. Thanks for clarifying.